pkg/server/server_test.go

```diff
@@ -362,18 +362,17 @@ func prepareFrontendConn(ctrl *gomock.Controller) *agentmock.MockAgentService_Co
 func prepareAgentConnMD(ctrl *gomock.Controller, proxyServer *ProxyServer) (*agentmock.MockAgentService_ConnectServer, Backend) {
 	// prepare the the connection to agent of proxy-server
 	agentConn := agentmock.NewMockAgentService_ConnectServer(ctrl)
-	agentID := uuid.New().String()
 	agentConnMD := metadata.MD{
 		":authority":       []string{"127.0.0.1:8091"},
-		"agentid":          []string{agentID},
+		"agentid":          []string{uuid.New().String()},
 		"agentidentifiers": []string{},
 		"content-type":     []string{"application/grpc"},
 		"user-agent":       []string{"grpc-go/1.42.0"},
 	}
 	agentConnCtx := metadata.NewIncomingContext(context.Background(), agentConnMD)
 	agentConn.EXPECT().Context().Return(agentConnCtx).AnyTimes()
-	backend := NewBackend(agentConn)
-	proxyServer.addBackend(agentID, backend)
+	backend, _ := NewBackend(agentConn)
```

Comment:
Even in a test I don't think its a good idea to ignore this error. I grant we may only log it but if we hit this error in a test run I do think we will want it to have been logged.